{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "656d7a20",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "\n",
    "adata = sc.read_h5ad('/labs/Aguiar/SSPA_BRAY/BRay/miscc/ctrl_sspa_test/Bcell_GEX_20251201_control.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cdefabce",
   "metadata": {},
   "outputs": [],
   "source": [
    "covid_adata = sc.read_h5ad('/labs/Aguiar/SSPA_BRAY/BRay/BRAY_FileTransfer/Covid19/gm_ctrl_covid.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3dc71110",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 2049 × 9400\n",
       "    obs: 'group', 'condition', 't2dm', 'cvda', 'pt_id', 'comb_id', 'Sex', 'celltype_subclust'\n",
       "    var: 'gene_ids', 'feature_types', 'genome', 'mito', 'highly_variable', 'means', 'dispersions', 'dispersions_norm'\n",
       "    uns: 'cell_type_colors', 'cell_type_rank_genes_groups', 'dendrogram_cell_type', 'dendrogram_leiden', 'hvg', 'leiden', 'leiden_colors', 'leiden_rank_genes_groups', 'neighbors', 'pca', 'umap'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'\n",
       "    varm: 'PCs'\n",
       "    layers: 'raw'\n",
       "    obsp: 'connectivities', 'distances'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "abbe73ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 194 × 16408\n",
       "    obs: 'sampleID', 'majorType'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covid_adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6923dd52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>condition</th>\n",
       "      <th>t2dm</th>\n",
       "      <th>cvda</th>\n",
       "      <th>pt_id</th>\n",
       "      <th>comb_id</th>\n",
       "      <th>Sex</th>\n",
       "      <th>celltype_subclust</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAAGGTAAGTAGCAAT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>c0d0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A1</td>\n",
       "      <td>A0</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACAAAGCATCGTGGC-1</th>\n",
       "      <td>A</td>\n",
       "      <td>c0d0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A1</td>\n",
       "      <td>A0</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACCAACGTAGTCTGT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>c0d0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A1</td>\n",
       "      <td>A0</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGCGTTAGGTGAGCT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>c0d0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A1</td>\n",
       "      <td>A0</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGTGAAAGATAACGT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>c0d0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A1</td>\n",
       "      <td>A0</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGAGAGGTACATA-1</th>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A8</td>\n",
       "      <td>A8</td>\n",
       "      <td>Male</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGTTCAAGTGTCT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A4</td>\n",
       "      <td>A4</td>\n",
       "      <td>Female</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGTTTCTAAGCGT-1</th>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A8</td>\n",
       "      <td>A8</td>\n",
       "      <td>Male</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGCACCAGTTA-1</th>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A8</td>\n",
       "      <td>A8</td>\n",
       "      <td>Male</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGTCTAGTGAC-1</th>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A8</td>\n",
       "      <td>A8</td>\n",
       "      <td>Male</td>\n",
       "      <td>B Cell</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2049 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   group condition  t2dm  cvda pt_id comb_id     Sex  \\\n",
       "AAAGGTAAGTAGCAAT-1     A      c0d0     0     0    A1      A0  Female   \n",
       "AACAAAGCATCGTGGC-1     A      c0d0     0     0    A1      A0  Female   \n",
       "AACCAACGTAGTCTGT-1     A      c0d0     0     0    A1      A0  Female   \n",
       "AAGCGTTAGGTGAGCT-1     A      c0d0     0     0    A1      A0  Female   \n",
       "AAGTGAAAGATAACGT-1     A      c0d0     0     0    A1      A0  Female   \n",
       "...                  ...       ...   ...   ...   ...     ...     ...   \n",
       "TTTGGAGAGGTACATA-1     A       NaN     0     0    A8      A8    Male   \n",
       "TTTGGTTCAAGTGTCT-1     A       NaN     0     0    A4      A4  Female   \n",
       "TTTGGTTTCTAAGCGT-1     A       NaN     0     0    A8      A8    Male   \n",
       "TTTGTTGCACCAGTTA-1     A       NaN     0     0    A8      A8    Male   \n",
       "TTTGTTGTCTAGTGAC-1     A       NaN     0     0    A8      A8    Male   \n",
       "\n",
       "                   celltype_subclust  \n",
       "AAAGGTAAGTAGCAAT-1            B Cell  \n",
       "AACAAAGCATCGTGGC-1            B Cell  \n",
       "AACCAACGTAGTCTGT-1            B Cell  \n",
       "AAGCGTTAGGTGAGCT-1            B Cell  \n",
       "AAGTGAAAGATAACGT-1            B Cell  \n",
       "...                              ...  \n",
       "TTTGGAGAGGTACATA-1            B Cell  \n",
       "TTTGGTTCAAGTGTCT-1            B Cell  \n",
       "TTTGGTTTCTAAGCGT-1            B Cell  \n",
       "TTTGTTGCACCAGTTA-1            B Cell  \n",
       "TTTGTTGTCTAGTGAC-1            B Cell  \n",
       "\n",
       "[2049 rows x 8 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "874c6bac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleID</th>\n",
       "      <th>majorType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S-HC008</td>\n",
       "      <td>CD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S-HC008</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S-HC008</td>\n",
       "      <td>CD8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S-HC008</td>\n",
       "      <td>Mono</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S-HC008</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>S-HC025</td>\n",
       "      <td>CD8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>S-HC025</td>\n",
       "      <td>Mono</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>S-HC025</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>S-HC025</td>\n",
       "      <td>DC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>S-HC025</td>\n",
       "      <td>Mega</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sampleID majorType\n",
       "0    S-HC008       CD4\n",
       "1    S-HC008        NK\n",
       "2    S-HC008       CD8\n",
       "3    S-HC008      Mono\n",
       "4    S-HC008         B\n",
       "..       ...       ...\n",
       "189  S-HC025       CD8\n",
       "190  S-HC025      Mono\n",
       "191  S-HC025         B\n",
       "192  S-HC025        DC\n",
       "193  S-HC025      Mega\n",
       "\n",
       "[194 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covid_adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "19a44e6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "      <th>mito</th>\n",
       "      <th>highly_variable</th>\n",
       "      <th>means</th>\n",
       "      <th>dispersions</th>\n",
       "      <th>dispersions_norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000188976</th>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.198184</td>\n",
       "      <td>1.161064</td>\n",
       "      <td>0.136665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000187608</th>\n",
       "      <td>ENSG00000187608</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>0.611551</td>\n",
       "      <td>2.112126</td>\n",
       "      <td>0.962336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186891</th>\n",
       "      <td>ENSG00000186891</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.082398</td>\n",
       "      <td>1.352114</td>\n",
       "      <td>0.490153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000078808</th>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.423276</td>\n",
       "      <td>1.163203</td>\n",
       "      <td>-0.941575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176022</th>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.046195</td>\n",
       "      <td>0.946936</td>\n",
       "      <td>-0.259522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000198840</th>\n",
       "      <td>ENSG00000198840</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>6.281722</td>\n",
       "      <td>6.465203</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000212907</th>\n",
       "      <td>ENSG00000212907</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>2.628459</td>\n",
       "      <td>2.866456</td>\n",
       "      <td>-0.830450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000198886</th>\n",
       "      <td>ENSG00000198886</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>5.686319</td>\n",
       "      <td>5.669989</td>\n",
       "      <td>-0.707107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000198786</th>\n",
       "      <td>ENSG00000198786</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>4.288185</td>\n",
       "      <td>4.347731</td>\n",
       "      <td>-1.049132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000198695</th>\n",
       "      <td>ENSG00000198695</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>1.396438</td>\n",
       "      <td>2.119126</td>\n",
       "      <td>-0.280344</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9400 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        gene_ids    feature_types  genome   mito  \\\n",
       "ENSG00000188976  ENSG00000188976  Gene Expression  GRCh38  False   \n",
       "ENSG00000187608  ENSG00000187608  Gene Expression  GRCh38  False   \n",
       "ENSG00000186891  ENSG00000186891  Gene Expression  GRCh38  False   \n",
       "ENSG00000078808  ENSG00000078808  Gene Expression  GRCh38  False   \n",
       "ENSG00000176022  ENSG00000176022  Gene Expression  GRCh38  False   \n",
       "...                          ...              ...     ...    ...   \n",
       "ENSG00000198840  ENSG00000198840  Gene Expression  GRCh38   True   \n",
       "ENSG00000212907  ENSG00000212907  Gene Expression  GRCh38   True   \n",
       "ENSG00000198886  ENSG00000198886  Gene Expression  GRCh38   True   \n",
       "ENSG00000198786  ENSG00000198786  Gene Expression  GRCh38   True   \n",
       "ENSG00000198695  ENSG00000198695  Gene Expression  GRCh38   True   \n",
       "\n",
       "                 highly_variable     means  dispersions  dispersions_norm  \n",
       "ENSG00000188976            False  0.198184     1.161064          0.136665  \n",
       "ENSG00000187608             True  0.611551     2.112126          0.962336  \n",
       "ENSG00000186891            False  0.082398     1.352114          0.490153  \n",
       "ENSG00000078808            False  0.423276     1.163203         -0.941575  \n",
       "ENSG00000176022            False  0.046195     0.946936         -0.259522  \n",
       "...                          ...       ...          ...               ...  \n",
       "ENSG00000198840            False  6.281722     6.465203          1.000000  \n",
       "ENSG00000212907            False  2.628459     2.866456         -0.830450  \n",
       "ENSG00000198886            False  5.686319     5.669989         -0.707107  \n",
       "ENSG00000198786            False  4.288185     4.347731         -1.049132  \n",
       "ENSG00000198695            False  1.396438     2.119126         -0.280344  \n",
       "\n",
       "[9400 rows x 8 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c8ad19b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000101811</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000004142</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000283992</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000112619</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143786</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000157077</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000152942</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000167232</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000157978</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188906</th>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16408 rows × 0 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: [ENSG00000101811, ENSG00000004142, ENSG00000283992, ENSG00000112619, ENSG00000143786, ENSG00000177943, ENSG00000138604, ENSG00000170085, ENSG00000182544, ENSG00000159216, ENSG00000157426, ENSG00000053702, ENSG00000230873, ENSG00000204978, ENSG00000102032, ENSG00000160959, ENSG00000163216, ENSG00000082146, ENSG00000102271, ENSG00000153395, ENSG00000167258, ENSG00000166035, ENSG00000122678, ENSG00000155367, ENSG00000198586, ENSG00000110717, ENSG00000146535, ENSG00000106018, ENSG00000100473, ENSG00000050628, ENSG00000119669, ENSG00000131389, ENSG00000242265, ENSG00000159917, ENSG00000159784, ENSG00000128617, ENSG00000168026, ENSG00000153002, ENSG00000139291, ENSG00000105991, ENSG00000146063, ENSG00000066468, ENSG00000146802, ENSG00000111906, ENSG00000143178, ENSG00000129559, ENSG00000137309, ENSG00000100078, ENSG00000125384, ENSG00000157240, ENSG00000198612, ENSG00000256892, ENSG00000135862, ENSG00000158578, ENSG00000198523, ENSG00000170293, ENSG00000179335, ENSG00000166228, ENSG00000196368, ENSG00000131116, ENSG00000156234, ENSG00000067177, ENSG00000176261, ENSG00000229833, ENSG00000109111, ENSG00000147650, ENSG00000241878, ENSG00000214694, ENSG00000105486, ENSG00000148965, ENSG00000011332, ENSG00000187867, ENSG00000127884, ENSG00000204531, ENSG00000165168, ENSG00000089195, ENSG00000165689, ENSG00000184588, ENSG00000099800, ENSG00000162068, ENSG00000197345, ENSG00000183624, ENSG00000026025, ENSG00000275111, ENSG00000183690, ENSG00000104299, ENSG00000109458, ENSG00000090621, ENSG00000179943, ENSG00000177125, ENSG00000141867, ENSG00000213853, ENSG00000100385, ENSG00000148343, ENSG00000130313, ENSG00000164053, ENSG00000182132, ENSG00000132182, ENSG00000162873, ENSG00000127084, ...]\n",
       "\n",
       "[16408 rows x 0 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covid_adata.var"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bray_cpu",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
